### PR TITLE
Fix #2212: Fix printing of optimizer hints

### LINKED ISSUE #2212: IRTreePrinter prints closing paren in front of class name
Culprit is this (refactored) `if`:
https://github.com/scala-js/scala-js/blob/master/ir/src/main/scala/org/scalajs/core/ir/Printers.scala#L758-L760

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

File: ir/src/main/scala/org/scalajs/core/ir/Printers.scala [hunk 1]
```diff
@@ -754,10 +754,7 @@ object Printers {
 
         case tree: ClassDef =>
           val ClassDef(name, kind, superClass, interfaces, jsName, defs) = tree
-          if (tree.optimizerHints != OptimizerHints.empty)
-            print("@hints(")
-            print(tree.optimizerHints.bits)
-            print(") ")
+          print(tree.optimizerHints)
```
Comment (sjrd) [inline ir/src/main/scala/org/scalajs/core/ir/Printers.scala]: Fail! ^^

File: ir/src/main/scala/org/scalajs/core/ir/Printers.scala [hunk 2]
```diff
@@ -895,6 +888,15 @@ object Printers {
     protected def print(c: Int): Unit =
       out.write(c)
 
+    protected def print(optimizerHints: OptimizerHints)(
+        implicit dummy: DummyImplicit): Unit = {
+      if (optimizerHints != OptimizerHints.empty) {
+        print("@hints(")
+        print(optimizerHints.bits)
```
Comment (sjrd) [inline ir/src/main/scala/org/scalajs/core/ir/Printers.scala]: Hum that's weird too, actually. Isn't it calling the `print(c: Int)` just above? There's probably a `.toString()` missing.

**Comment by sjrd** [comment]
That's all.

File: ir/src/main/scala/org/scalajs/core/ir/Printers.scala (hunk 2 above)
Comment (gzm0) [inline ir/src/main/scala/org/scalajs/core/ir/Printers.scala]: Yep:

```
@hints() def productionMode__Z(): boolean
```

**Comment by gzm0** [comment]
Updated. Tested locally.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**